src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -71,11 +73,16 @@ Napi::Value CoreWrap::read_model_async(const Napi::CallbackInfo& info) {
 }
 
 Napi::Value CoreWrap::compile_model_sync(const Napi::CallbackInfo& info,
-                                                const Napi::Object& model,
-                                                const Napi::String& device) {
-    auto m = Napi::ObjectWrap<ModelWrap>::Unwrap(model);
-    const auto& compiled_model = _core.compile_model(m->get_model(), device);
-    return CompiledModelWrap::Wrap(info.Env(), compiled_model);
+                                         const Napi::Object& model,
+                                         const Napi::String& device) {
+    if (model.InstanceOf(info.Env().GetInstanceData<AddonData>()->model_prototype->Value().As<Napi::Function>())) {
```

Comment:
```suggestion
    const auto model_prototype = info.Env().GetInstanceData<AddonData>()->model_prototype;
    if (model_prototype && model.InstanceOf(model_prototype->Value().As<Napi::Function>())) {
```
Assert also that `GetInstanceData<AddonData>()` and `Value()` provide valid accesses.